docs/registry/README.md

```diff
@@ -0,0 +1,193 @@
+# Registry Proposal
+
+## Objective
+
+Design an **initial** version of the Registry that can support discoverability of 
+the different event types that can be consumed from the eventing mesh.
+
+## Requirements
+
+Our design revolves around the following core requirements:
+
+1. We should have a Registry per namespace to enforce isolation.
+2. The Registry should contain the event types that can be consumed from
+the eventing mesh. If an event type is not ready for consumption, we should explicitly indicate so.
+3. The event types stored in the Registry should contain all the required information 
+for a consumer to create a Trigger without resorting to some other OOB mechanism.
+
+## Design Ideas
+
+### EventType CRD
+
+We propose having a namespaced EventType CRD. Here is an example of how a CR would look like:
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  name: repopush
+spec:
+  type: repo:push
+  source: my-user/my-repo
+  schema: my-schema
+  broker: default
+```
+
+
+- The `name` of the EventType is advisory, non-authoritative. Given that Cloud Event types can 
+contain characters that may not comply with Kubernetes naming conventions, we will (slightly) 
+modify those names to make them K8s-compliant, whenever we need to generate them. 
+
+- `type` is authoritative. This refers to the Cloud Event type as it enters into the eventing mesh. 
+
+- `source`: an identifier of where we receive the event from. This might not necessarily be the Cloud Event source 
+attribute. If we receive the event from our receive adaptors, the info might come in a Cloud Event custom extension (e.g., from).
+
+- `schema` is a URI with the EventType schema. It may be a JSON schema, a protobuf schema, etc. It is optional.
+
+- `broker` refers to the Broker that can provide the EventType. 
+
+### EventType Instantiation
+
+We foresee the following ways of populating the Registry:
+
+**1. Event Source CR installation**
+
+Upon installation of an Event Source CR, the source will register its EventTypes.
```

Comment:
I think this is just the case for the GH source, to sorta "filter" the events to "provide"

Reply:
GH is one source that potentially has many types, and we know those types in advance.
But for the other sources, where we know what event types will be sending, IMO we should go ahead and add them to the registry.
For example, AWSSQS always sends one eventType (`aws.sqs.message`), cronjob as well (`dev.knative.cronjob.event`). Also GCPPubSub (`google.pubsub.topic.publish`), and kafka (`dev.knative.kafka.event`). 
When we instantiate a particular CR (say kafka with topic1 and topic2), we will populate the registry with two EventTypes: 1) type: `dev.knative.kafka.event`, source: `topic1`, and 2) type: `dev.knative.kafka.event`, source: `topic2`.

Does it make sense?